### PR TITLE
Remove timestamps from docs/api-reference/*/*.html

### LINKED ISSUE #46814: docs/api-reference/*/*.html contains a timestamp
This means any 2 PRs that touch this fine CAN'T POSSIBLY not conflict.  We should just strip the timestamp.

`sed 's/Last updated ....-..-.. ..:..:.. ...//'`

easy change, I hope, just needs to find the right back-and-forth in `hack/lib/swagger.sh` `kube::swagger::gen_api_ref_docs`

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

If there are 2 or more PR(s) in the queue, they will end up with
conflicts (and rechecks). So let us remove the timestamp entirely
when we generate the files.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Fixes #46814

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/lib/swagger.sh [hunk 1]
```diff
@@ -131,9 +131,12 @@ kube::swagger::gen_api_ref_docs() {
     if [[ -e "${output_dir}/${file}" && -e "${output_tmp}/${file}" ]]; then
       echo "comparing ${output_dir}/${file} with ${output_tmp}/${file}"
 
+      # Remove the timestamp to reduce conflicts in PR(s)
+      sed -i 's/^Last updated.*$//' "${output_tmp}/${file}"
```
Comment (cblecker) [inline hack/lib/swagger.sh]: If we're using sed, we should use something like https://github.com/kubernetes/kubernetes/blob/master/staging/copy.sh#L64-L71 to ensure that we don't have flag issues between macOS/Linux

**Review by cblecker — CHANGES_REQUESTED** [review]

**Comment by fejta** [comment]
/unassign
/assign @cblecker 

Please add me back if my help is needed!

**Comment by cblecker** [comment]
This looks good, although please revert the change to `api/openapi-spec/swagger.json`. We shouldn't be downgrading the spec version here.

**Comment by cblecker** [comment]
/lgtm

/unassign @brendandburns 
/assign @thockin 

@thockin As this is a docs improvement and small change to a single hack/ script, do you think this should be merged for 1.7?

**Comment by dims** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

File: hack/lib/swagger.sh [hunk 2]
```diff
@@ -127,13 +127,25 @@ kube::swagger::gen_api_ref_docs() {
...
+    echo "Failed to find GNU sed as sed or gsed. If you are on Mac: brew install gnu-sed." >&2
+    exit 1
+  fi
+
   while read file; do
     if [[ -e "${output_dir}/${file}" && -e "${output_tmp}/${file}" ]]; then
       echo "comparing ${output_dir}/${file} with ${output_tmp}/${file}"
 
+      # Remove the timestamp to reduce conflicts in PR(s)
+      $SED -i 's/^Last updated.*$//' "${output_tmp}/${file}"
```
Comment (thockin) [inline hack/lib/swagger.sh]: How sure are we that this is precise enough?  Maybe "^Last updated.*UTC$" ?

**Comment by thockin** [comment]
AWESOME.  YOU ARE A HERO.

**Comment by thockin** [comment]
As soon as you rebase this and get a green test run, we should merge ASAP.  Everyone will have to rebase one last time.

**Comment by thockin** [comment]
@dchen1107 FYI - this will make things move faster after the initial pain.

@dims - poke me by slack or something when good

**Comment by dchen1107** [comment]
@dims Can you rebase this pr? Thanks!

**Comment by dims** [comment]
yes of course @dchen1107 coming up in a few

**Comment by cblecker** [comment]
Failing verify because of some new lines in the diff.

Is it possible to split this out into two commits.. one with the changes to `hack/lib/swagger.sh`, and one with the `make update` after that change? Might make rebasing easier.

**Comment by dims** [comment]
@cblecker one last try and then i'll split things up

**Comment by cblecker** [comment]
@dims if it passes this time, then I'm fine with just leaving it as one and getting it merged. If you need to go back and make changes anyways because verify fails again, then yeah, two commits is easier to review :)

Thanks for this change.. excited to get it in!

**Comment by dims** [comment]
@cblecker thanks. sorry, on a plane, so just doing the minimum and using the CI to run stuff :)

**Comment by cblecker** [comment]
@dims No worries -- I've been there ✈️! As soon as verify goes green, I'll LGTM this :)

**Comment by dims** [comment]
regular one worked, next stop - federation

**Comment by dims** [comment]
yay, passed verify

**Comment by cblecker** [comment]
Commenting for gubernator to update this PR's status :)

**Pull request merged**